lib/gds_api/test_helpers/account_api.rb

```diff
@@ -82,6 +82,37 @@ def stub_account_api_unauthorized_user_info(**options)
         )
       end
 
+      ##################
+      # DELETE /api/user
+      ##################
+
+      def stub_account_api_success_delete_account(subject_identifier:, **options)
+        stub_account_api_request(
+          :delete,
+          "/api/user/#{subject_identifier}",
+          response_status: 204,
+          **options,
+        )
+      end
+
+      def stub_account_api_not_found_delete_account(subject_identifier:, **options)
+        stub_account_api_request(
+          :delete,
+          "/api/user/#{subject_identifier}",
+          response_status: 404,
+          **options,
+        )
+      end
+
+      def stub_account_api_unauthorized_delete_account(subject_identifier:, **options)
```

Comment:
I think you can skip this case, as if you get a 401 you're using the wrong signon token